src/main/java/com/peerrequest/app/api/EntriesController.java

```diff
@@ -0,0 +1,81 @@
+package com.peerrequest.app.api;
+
+import com.peerrequest.app.model.Category;
+import com.peerrequest.app.model.Document;
+import com.peerrequest.app.model.Entry;
+import com.peerrequest.app.model.User;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+@RestController
+@RequestMapping
+public class EntriesController {
+
+    @GetMapping(value = "categories/{categoryId}/entries",
+            produces = "application/json")
+    public List<Entry> getEntries(@PathVariable final int categoryId) {
+        return List.of(mockEntry());
+    }
+
+    @GetMapping(value = "categories/{categoryId}/entries/{entryId}",
+            produces = "application/json")
+    public Entry getEntry(@PathVariable final int categoryId,
+                          @PathVariable final int entryId) {
+        return mockEntry();
+    }
+
+    @GetMapping(value = "categories/{categoryId}/entries/{entryId}/paper",
+            produces = "application/json")
+    public Document getResearchPaper(@PathVariable final int categoryId,
+                                     @PathVariable final int entryId) {
+        return mockEntry().getDocument();
+    }
+
+    @PostMapping(value = "categories/{categoryId}/entries",
+            consumes = "application/json",
+            produces = "application/json")
+    public Entry postEntry(@PathVariable final int categoryId,
+                             @RequestBody final Entry entry) {
+        return mockEntry();
+    }
+
+    @PatchMapping(value = "categories/{categoryId}/entries/{entryId}",
+            consumes = "application/json",
+            produces = "application/json")
+    public Entry patchEntry(@PathVariable final int categoryId,
+                                    @PathVariable final int entryId,
+                                    @RequestBody final Entry.EntryUpdater entryUpdater) {
+        Entry updatedEntry = mockEntry();
+        updatedEntry.setAuthors("Karol Laseconde");
+        updatedEntry.setTitle("Updated Title");
+        return updatedEntry;
```

Comment:
doesn't update the hardcoded entry. Maybe updating the hardcoded entry is not needed. Evaluate by yourself.